src/api/app/components/sourcediff_component.rb

```diff
@@ -41,7 +41,10 @@ def target_package
   def diff_list(sourcediff)
     files = sourcediff['files'].sort_by { |k, _v| sourcediff['filenames'].find_index(k) }.to_h
 
-    files.each_with_index do |(filename, _contents), file_index|
+    files.each_with_index do |(filename, contents), file_index|
+      files[filename]['disabled'] = contents['state'].in?(%w[deleted added]) && contents['diff']['_content'].nil?
```

Comment:
Why only deleted/added?

Reply:
I’ve only encountered three states: added, deleted, and changed. For added or deleted files, we don’t receive any diff content. In the case of `changed` files, we receive the number of lines modified.

Reply:
Hm okay, sounds like another feature then? Because why wouldn't I see and all green/all red diff for files added/removed?

Reply:
Hm and isn't that the `missingok:bool?` backend parameter?

Reply:
Right now, added / removed binary files do not show diffs, except for tarballs, that show content changes, regular files show full diffs, even when added or removed, as you can see in the attached screenshot
<img width="1865" height="2514" alt="Screenshot 2025-08-05 at 09-33-31 Request 20 - Open Build Service" src="https://github.com/user-attachments/assets/535f14f5-0fa1-43f8-89d9-854e3d812fe2" />